server/jetstream_cluster.go

```diff
@@ -7295,23 +7295,29 @@ func (s *Server) jsClusteredMsgDeleteRequest(ci *ClientInfo, acc *Account, mset
 }
 
 func encodeAddStreamAssignment(sa *streamAssignment) []byte {
+	csa := *sa
```

Comment:
We are copying the fields here we will throw away, can we be a bit smarter here and avoid unecessary allocations and copies?

Reply:
In the case of `ca` and `sa`, all the fields are needed, the only one that gets overwritten is `Client`. We could pool them, since annoyingly every use of `json.Marshal(...)` or `json.NewEncoder(...).Encode(...)` will result in a heap escape...

Reply:
ok let's attempt that change.

If we hold the js lock so that the sa or ca can not be changed, we could capture ci, nil it out of original before the csa := *sa and then put it back. Can also use the temp vairable in the call ro function to return stripped down version for proposals.